### PR TITLE
Fix suid detection and instance boot

### PR DESCRIPTION
**Pull request by cclerget**
**Description of the Pull Request (PR):**

- Fix SUID detection
- Add missing UTS option for instance.start
- Fix privblock tests and getuid for TARGET_UID/TARGET_GID

Changelog has already been updated for boot/start instance feature

**Checkoff for all PRs:**

- [x] I have read the [Guidelines for Contributing](https://github.com/singularityware/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [ ] I have added changes to the [CHANGELOG](https://github.com/singularityware/singularity/blob/development/CHANGELOG.md) and and documentation updates to the [singularityware](https://www.github.com/singularityware/singularityware.github.io) documentation base.
- [x] I have tested this PR locally with a `make test`
- [x] This PR is NOT against the project's master branch
- [ ] I have added myself as a contributor to the [contributors's file](https://github.com/singularityware/singularity/blob/master/CONTRIBUTORS.md)
- [x] This PR is ready for review and/or merge

Attn: @singularityware-admin

### CODE REVIEW

File: src/lib/runtime/files/hostname/hostname.c [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    if ( hostname_fd == NULL ) {
+        singularity_message(ERROR, "Couldn't create hostname session file\n");
+        ABORT(255);
+    }
+
+    if ( strlen(hostname) > HOST_NAME_MAX ) {
+        hostname[HOST_NAME_MAX] = '\0';
+    }
+
+    fprintf(hostname_fd, "%s\n", hostname);
```
Comment (gmkurtzer) [inline src/lib/runtime/files/hostname/hostname.c:64]: I am pretty sure that we need to set the hostname directly via systemcall, not via file and count on systemd/SysVinit/upstart/pick_your_favorite_init or distribution to do it all the same.

File: src/lib/runtime/ns/uts/uts.c [hunk 2]
```diff
@@ -49,6 +60,10 @@ int _singularity_runtime_ns_uts(void) {
         singularity_message(ERROR, "Could not virtualize UTS namespace: %s\n", strerror(errno));
         ABORT(255);
     }
+    if ( hostname && sethostname(hostname, hostname_len) < 0 ) {
```
Comment (gmkurtzer) [inline src/lib/runtime/ns/uts/uts.c:63]: Ahh, I see you are doing it in both ways. I think we should prefer to do it only via systemcall.

**Review by gmkurtzer — CHANGES_REQUESTED** [review]

File: src/lib/runtime/files/hostname/hostname.c (hunk 1 above)
Comment (cclerget) [reply src/lib/runtime/files/hostname/hostname.c:64]: using only syscall won't work when --boot option is given, init systems set hostname based on /etc/hostname content (container default is often localhost.localdomain), so both are required to handle this case. Or just had a check : if boot requested set and bind /etc/hostname. Make me think when no hostname is given while starting instance to set it with instance name by default, no ?

**Review by gmkurtzer — APPROVED** [review]

**Comment by gmkurtzer** [comment]
Thank you @cclerget!

**Pull request merged**